pyvista/plotting/scalar_bars.py

```diff
@@ -28,12 +28,34 @@ def __repr__(self):
         """Nice representation of this class."""
         lines = []
         lines.append('Scalar Bar Title     Interactive')
-        for title, actor in self._scalar_bar_actors.items():
+        for title in self._scalar_bar_actors:
             interactive = title in self._scalar_bar_widgets
             title = f'"{title}"'
             lines.append(f'{title:20} {str(interactive):5}')
         return '\n'.join(lines)
 
+    def _remove_mapper_from_plotter(self, actor, reset_camera=False, render=False):
+        """Remove an actor's mapper from the given plotter's _scalar_bar_mappers."""
+        try:
+            mapper = actor.GetMapper()
+        except AttributeError:
+            return
+        for name in list(self._scalar_bar_mappers.keys()):
```

Comment:
Nitpick, but I'd just call `list` on the dict itself:
```suggestion
        for name in list(self._scalar_bar_mappers):
```

Reply:
Cleaner.  Didn't realize that `list(dict)` returns a list of keys.

We can go one step further and actually just use:
```
for name in self._scalar_bar_mappers:
```
Since it will iterate through the keys in the dict.

Reply:
Turns out I can just change your suggestion :)

Reply:
No, because we might end up deleting items from the list, which is forbidden in newer python (for good reason). We do need a copy for this.